core/internal/cltest/client.go

```diff
@@ -258,7 +258,23 @@ func (c *SimulatedBackendClient) SubscribeNewHead(ctx context.Context, channel c
 			if h == nil {
 				channel <- nil
 			} else {
-				channel <- &models.Head{Number: h.Number.Int64(), Hash: NewHash()}
+				func() {
+					head := &models.Head{Number: h.Number.Int64(), Hash: NewHash()}
+					// Catch a potential panic and print a warning about it.
+					//
+					// Seems that chainlink can shut itself down before the simulated
+					// backend is done sealing blocks, leading to a panic when this
+					// handler tries to push information about a new head onto a closed
+					// channel.
+					defer func() {
+						if r := recover(); r != nil {
```

Comment:
Seems like catching a panic here is just "sweeping the problem under the rug", the bug here seems to be that we're writing to a closed channel and rescuing the panic doesn't fix the bug. What's making it difficult to fix SubscribeToNewHead?

Reply:
Thanks for the suggestion. I have rewritten this to plug into the underlying `ethereum.Subscription`, in the latest commit.